crypto/async/arch/async_posix.h

```diff
@@ -25,17 +25,33 @@
 #  define ASYNC_POSIX
 #  define ASYNC_ARCH
 
+#  ifdef __CET__
+/*
+ * When Intel CET is enabled, makecontext will create a different
+ * shadow stack for each context.  async_fibre_swapcontext cannot
+ * use _longjmp.  It must call swapcontext to swap shadow stack as
+ * well as normal stack.
+ */
+#    define USE_SWAPCONTEXT
+#  endif
 #  include <ucontext.h>
-#  include <setjmp.h>
+#  ifndef USE_SWAPCONTEXT
+#    include <setjmp.h>
+#  endif
 
 typedef struct async_fibre_st {
     ucontext_t fibre;
+#    ifndef USE_SWAPCONTEXT
     jmp_buf env;
     int env_init;
+#    endif
 } async_fibre;
 
 static ossl_inline int async_fibre_swapcontext(async_fibre *o, async_fibre *n, int r)
```

Comment:
Side note: As far as I can tell the `r` parameter is unused. Commit 7070e5ca2 originally changed the code from using `swapcontext` to `longjmp` on the basis of performance . The code used to just call `swapcontext` like you are changing it back to here - except it did something different in the case of `r == 0`. However, as far as i can tell *all* places where we call `async_fibre_swapcontext` set r to 1 - so presumbably this could get refactored away at some point.